odoo/cli/populate.py

```diff
@@ -43,17 +55,40 @@ def populate(cls, env, size, model_patterns=False):
 
             _logger.log(25, 'Populating database')
             for model in ordered_models:
+                if profiling_enabled:
+                    profiling_context = odoo.tools.profiler.Profiler(
+                        description=f'{model} {size}',
+                        db=env.cr.dbname
+                    )
+                else:
+                    profiling_context = nullcontext()
+
+                if not commit:
+                    commit_context = patch('odoo.sql_db.Cursor.commit')
+                else:
+                    commit_context = nullcontext()
+
                 _logger.info('Populating database for model %s', model._name)
                 t0 = time.time()
-                registry.populated_models[model._name] = model._populate(size).ids
+                with profiling_context, commit_context:
+                    registry.populated_models[model._name] = model._populate(size).ids
                 # todo indicate somewhere that model is populated
-                env.cr.commit()
+
+                # force the flush to make sure population time still
+                # considers flushing all values to database
+                env['base'].flush()
```

Comment:
should be done in the profiling context to make sure queries are in the speedscope.